dnstable_convert.c

```diff
@@ -310,11 +310,19 @@ process_rdata_slice(Nmsg__Sie__DnsDedupe *dns, size_t i, ubuf *key, ubuf *val)
 static void
 process_rdata_name_rev(Nmsg__Sie__DnsDedupe *dns, size_t i, ubuf *key, ubuf *val)
 {
-	size_t offset;
+	size_t offset, len = dns->rdata[i].len;
 	uint8_t name[WDNS_MAXLEN_NAME];
 	wdns_res res;
 
 	switch (dns->rrtype) {
+	case WDNS_TYPE_SOA:
+		res = wdns_len_uname(dns->rdata[i].data,
+				     dns->rdata[i].data + len,
```

Comment:
you are using len before the check on its validity that is at line 344

Reply:
I do not think a validity check (effectively, `if (len == 0) return;` would add to the readability. We would merely be passing `wdns_len_uname()` a zero-length buffer, a case which it is expected to handle (and does).

Reply:
I hear you, but we have such a check later 
```C
if (dns->rdata[i].len == 0 || dns->rdata[i].len <= offset)
		return;
```

Reply:
Added check.